block-producer/src/txqueue/mod.rs

```diff
@@ -100,15 +101,35 @@ where
             locked_ready_queue.drain(..).collect()
         };
 
-        let tx_groups = txs.chunks(self.options.batch_size).map(|txs| txs.to_vec());
+        while !txs.is_empty() {
+            let mut batch = Vec::with_capacity(self.options.batch_size);
+            let mut notes_in_batch = 0;
+
+            while let Some(tx) = txs.pop() {
+                notes_in_batch += tx.output_notes().num_notes();
+
+                debug_assert!(
+                    tx.output_notes().num_notes() <= MAX_NOTES_PER_BATCH,
+                    "Sanity check, the number of output notes of a single transaction must never be larger than the batch maximum",
+                );
```

Comment:
The real limit is much lower, defined by:

https://github.com/0xPolygonMiden/miden-base/blob/353589bc20c41db8138e11d6b5eecb419e693043/objects/src/constants.rs#L19-L20

This is just a sanity check. That's why it is only a `debug_assert`